pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -196,10 +223,133 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 			return reconcile.Result{}, err
 		}
 	}
+	addressSpaceURL := ""
+	//Begin AMQ resource reconcile
+	if instance.Spec.UseMessageBroker {
+		//check that address space exists
+		foundAddressSpace := newAddressSpace(instance)
+
+		// Set UnifiedPushServer instance as the owner and controller
+		if err := controllerutil.SetControllerReference(instance, foundAddressSpace, r.scheme); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: foundAddressSpace.Name, Namespace: foundAddressSpace.Namespace}, foundAddressSpace)
+		if err != nil && errors.IsNotFound(err) {
+			reqLogger.Info("Creating a new Address Space", "AddressSpace.Namespace", foundAddressSpace.Namespace, "AddressSpace.Name", foundAddressSpace.Name)
+			err = r.client.Create(context.TODO(), foundAddressSpace)
```

Comment:
Let me make sure I'm following you;

1. addressSpace is the object we want to have
2. foundAddressSpace is an object that possibly exists inside of k8s 
2.a If there is a IsNotFound error getting foundAddressSpace then use addressSpace to create a new addressSpace
2.b otherwise check the status of foundAddressSpace and continue or requeue as necessary.

I see the problems you are talking about with the names being shit ;)